control-plane/connect-inject/endpoints_controller.go

```diff
@@ -88,7 +87,7 @@ type EndpointsController struct {
 	// ConsulClientConfig is the config for the Consul API client.
 	ConsulClientConfig *consul.Config
 	// ConsulServerConnMgr is the watcher for the Consul server addresses.
-	ConsulServerConnMgr *discovery.Watcher
+	ConsulServerConnMgr consul.ServerConnectionManager
```

Comment:
All changes in the `connect-inject` package are to switch tests to use the mock of the `consul-server-connection-manager`.